src/media/ros/ros_writer.cpp

```diff
@@ -492,11 +492,17 @@ namespace librealsense
         }
     }
 
-    rs2_extension ros_writer::get_processing_block_extension(const std::shared_ptr<processing_block_interface> block)
+    static std::string get_processing_block_extension_name( const std::shared_ptr< processing_block_interface > block )
     {
-#define RETURN_IF_EXTENSION(E, T)\
-    if (Is<ExtensionToType<T>::type>(E))\
-    return T;\
+        // We want to write the block name (as opposed to the extension name):
+        // The block can behave differently and have a different name based on how it was created (e.g., the disparity
+        // filter). This makes new rosbag files incompatible with older librealsense versions.
```

Comment:
How about old files with new versions?

Reply:
We compare both extension names and block names. Old files will load into the new, I tested it.